packages/core/streams/ai-stream.ts

```diff
@@ -5,110 +5,142 @@ import {
-) {
-  const encoder = new TextEncoder()
-  let fullResponse = ''
-
+): TransformStream<string, Uint8Array> {
+  const textEncoder = new TextEncoder()
+  let aggregatedResponse = ''
   const { onStart, onToken, onCompletion } = callbacks || {}
 
-  return new TransformStream<string, Uint8Array>({
+  return new TransformStream({
     async start(): Promise<void> {
       if (onStart) await onStart()
     },
 
     async transform(message, controller): Promise<void> {
-      controller.enqueue(encoder.encode(message))
+      controller.enqueue(textEncoder.encode(message))
 
       if (onToken) await onToken(message)
-      if (onCompletion) fullResponse += message
+      if (onCompletion) aggregatedResponse += message
     },
 
     async flush(): Promise<void> {
-      await onCompletion?.(fullResponse)
+      if (onCompletion) await onCompletion(aggregatedResponse)
     }
   })
 }
 
-// If we're still at the start of the stream, we want to trim the leading
-// `\n\n`. But, after we've seen some text, we no longer want to trim out
-// whitespace.
-export function trimStartOfStreamHelper() {
-  let start = true
-  return (text: string) => {
-    if (start) text = text.trimStart()
-    if (text) start = false
+/**
+ * Creates a function to trim the start of a stream.
+ * @return {Function} A function that trims the start of a stream.
+ */
+export function trimStartOfStreamHelper(): (text: string) => string {
+  let isStreamStart = true
+
+  return (text: string): string => {
+    if (isStreamStart) {
+      text = text.trimStart()
+      if (text) isStreamStart = false
+    }
     return text
   }
 }
 
+/**
+ * Creates an AIStream from the response.
+ * @param {Response} response - The response.
+ * @param {AIStreamParser} customParser - The custom parser function.
+ * @param {AIStreamCallbacks} callbacks - The callbacks.
+ * @return {ReadableStream} The AIStream.
+ * @throws Will throw an error if the response is not OK.
+ */
 export function AIStream(
-  res: Response,
+  response: Response,
   customParser: AIStreamParser,
   callbacks?: AIStreamCallbacks
 ): ReadableStream {
-  // If the response is not OK, we want to throw an error to indicate that
-  // the AI service is not available.
-  // When catching this error, we can check the status code and return a handled
-  // error response to the client.
-  if (!res.ok) {
+  if (!response.ok) {
     throw new Error(
-      `Failed to convert the response to stream. Received status code: ${res.status}.`
+      `Failed to convert the response to stream. Received status code: ${response.status}.`
     )
   }
 
-  const stream =
-    res.body ||
-    new ReadableStream({
-      start(controller) {
-        controller.close()
-      }
-    })
+  const responseBodyStream = response.body || createEmptyReadableStream()
 
-  return stream
+  return responseBodyStream
     .pipeThrough(createEventStreamTransformer(customParser))
     .pipeThrough(createCallbacksTransformer(callbacks))
 }
+
+/**
```

Comment:
Remove changes to ai-stream, seems like mixing PRs here.

Reply:
My apologies, there was a mix-up in the local setup. I reverted it now.